### PR TITLE
#70 GUI for historical conversations

### PR DESCRIPTION
**Pull request by ExiRain**
- Added linkifier component
- Added linkifier dependecies
- Added decoding to historical message
- Added decoding for NewExampleModal
- Added decoding fro newResponseModal

This fix to properly display historical messages that might include characters that not meant to be displayed like %20 instead of whitespace.
[Related Task](https://github.com/buerokratt/Training-Module/issues/70).

### CODE REVIEW

**Review by KasparMinn — APPROVED** [review]

File: GUI/src/components/HistoricalChat/ChatMessage.tsx [hunk 1]
```diff
@@ -1,23 +1,31 @@
-import { FC } from 'react';
+import React, { FC } from 'react';
```
Comment (bartekpospiech) [inline GUI/src/components/HistoricalChat/ChatMessage.tsx:1]: **question:** do you need react here?

File: GUI/src/components/HistoricalChat/ChatMessage.tsx [hunk 2]
```diff
@@ -1,23 +1,31 @@
...
-      <time dateTime={message.authorTimestamp} className='historical-chat__message-date'>
-        {format(new Date(message.authorTimestamp), 'HH:ii:ss')}
-      </time>
-    </div>
-  );
+    return (
+        <div className="historical-chat__message">
+            <div
+                className="historical-chat__message-text"
+                onClick={onMessageClick ? () => onMessageClick(message) : undefined}
```
Comment (bartekpospiech) [inline GUI/src/components/HistoricalChat/ChatMessage.tsx:17]: **suggestion:** imo it's better to handle empty click agains `null` not `undefined`

File: GUI/src/components/HistoricalChat/index.tsx [hunk 3]
```diff
@@ -75,7 +68,7 @@ const HistoricalChat: FC<ChatProps> = ({ chat, trigger }) => {
       if(data.newIntent) {
         return addIntentWithExample({ intentName: data.intentName || '',newExamples:  data.example });
       }
-      return addExampleFromHistory(data.intent, { example: data.example });
+      return addExampleFromHistory(data.intentName || '', { example: data.example });
```
Comment (bartekpospiech) [inline GUI/src/components/HistoricalChat/index.tsx:71]: suggestion: `data.intentName ?? ''`

File: GUI/src/components/HistoricalChat/ChatMessage.tsx (hunk 2 above)
Comment (ExiRain) [reply GUI/src/components/HistoricalChat/ChatMessage.tsx:17]: Then ill get type complains

**Pull request merged**